### PR TITLE
[WIP] nlp.py updated for TF2, examples of NLP is pending

### PR DESCRIPTION
**Pull request by JingqingZ**
### Checklist
- [x] I've tested that my changes are compatible with the latest version of Tensorflow.
- [x] I've read the [Contribution Guidelines](https://github.com/tensorlayer/tensorlayer/blob/master/CONTRIBUTING.md)
- [x] I've updated the documentation if necessary.

### CODE REVIEW

**Comment by JingqingZ** [comment]
@1FengL Is this PR ready to review?

**Comment by 1FengL** [comment]
yes. text tutorial runnable, rnn api changed as required. PTB tutorial need
review as its perplexity is too high

Jingqing Zhang <notifications@github.com> 于 2019年5月12日周日 下午1:48写道：

**Comment by zsdonghao** [comment]
Hello, remember to put your name in `changelog.md` and use `yapf -i xxxx.py` to correct the code format.

We just have a discussion for the augments of RNN, we would like to change the :

- `return_last_state` equals to `True` by default.
- `return_2d` and `return_last_output` to `False` by default.

We are also thinking a better name for the augments ... feel free to discuss in the group chat.

**Comment by 1FengL** [comment]
ok, will do so. thanks

Hao <notifications@github.com> 于 2019年5月12日周日 下午2:15写道：

**Comment by zsdonghao** [comment]
It is time to merge this PR.

Please run `yapf -i xxx.py` or `yapf -i --recursive xxx(folder)xxx` to refactor the code format.

File: examples/text_generation/tutorial_generate_text.py [hunk 1]
```diff
@@ -222,166 +243,88 @@ def main_lstm_generate_text():
...
-                    lstm1.initial_state: state1
-                }
-            )
-            costs += _cost
-            iters += sequence_length
+            with tf.GradientTape() as tape:
+                ## compute outputs
+                logits, lstm_state = net(x, initial_state=lstm_state)
+                ## compute loss and update model
+                cost = tl.cost.cross_entropy(logits, tf.reshape(y, [-1]), name='train_loss')
```
Comment (zsdonghao) [inline examples/text_generation/tutorial_generate_text.py]: @JingqingZ it is the correct loss here??

Comment (zsdonghao) [reply examples/text_generation/tutorial_generate_text.py]: also, I saw the `max_max_epoch` is deleted, not sure whether is it the reason that @1FengL 's loss is different from the loss of the original implementation?

Comment (1FengL) [reply examples/text_generation/tutorial_generate_text.py]: Hi, the max_max_epoch is deleted in text generation tutorial for simplicity, should I add it back? The loss of text generation tutorial is fine and it is the ptb tutorial whose loss is abnormal.
